asmrun/fail.c

```diff
@@ -121,11 +121,21 @@ void caml_failwith (char const *msg)
   caml_raise_with_string((value) caml_exn_Failure, msg);
 }
 
+void caml_failwith_value (value msg)
```

Comment:
I don't think any `CAML*` annotation is required.  As far as I know, `CAMLexport` is now redundant, and `CAMLprim` is only used to mark primitive functions called from OCaml for bytecode compilation.  The function here is just a "normal" function to be called from C.